src/cortexpy/tesserae.py

```diff
@@ -264,17 +297,20 @@ def __termination(self, l1, pos_max, state_max, who_max):
 
             if self.maxpath_state[cp + 1] != 3:
                 pos_target -= 1
-        return cp
+        return cp, pos_max, who_max, state_max
 
-    def __recurrence(self, query, panel, lsize_l, l1, max_r, pos_max, state_max, who_max):
+    def __recurrence(self, query, panel, lsize_l, l1, max_r, pos_max, state_max, who_max, offset=0, l0=2, store_states=False):
```

Comment:
What do `l1` and `l0` stand for? I think it would be good to have some documentation on what some of the input parameters mean. Alternatively, perhaps encapsulating some of these parameters in an object might reduce the need for documentation.